### PR TITLE
BP-72 CTA modal for posting a question via the single business page

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: Bizyhood/Cta.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      $headers['Content-type'] = "application/json";
+      
+      $request = $this->submit_form($bizyhood_id, json_encode($params), $headers);
+      
+      wp_send_json($request);
+      
+    }
+    
+    
+    public static function submit_form($bizyhood_id, $params, $headers)
```
Comment (ghost) [inline Bizyhood/Cta.php]: Shouldn't this method go into the Api module to keep things consistent?

File: Bizyhood/Cta.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      $client = Bizyhood_oAuth::oAuthClient();
+      
+      if (is_wp_error($client)) {
+        return false;
+      }
+
+      try {
+        $response = $client->fetch($api_url . "/v2/business/" . $bizyhood_id.'/topic/', $params, $client::HTTP_METHOD_POST, $headers);        
+      } catch (Exception $e) {
+        return false;
```
Comment (ghost) [inline Bizyhood/Cta.php]: We'd need to know what the exception is. Didn't we have logging setup in the client?

File: Bizyhood/Public/js/bizyhood-custom.js [hunk 3]
```diff
@@ -1,4 +1,59 @@
...
+        if(response.error == true) {
+          jQuery('.response_area').addClass('bh_error').html(response.message);
+          
+          setTimeout(function () {
+            jQuery('.response_area').removeClass('bh_error').html('');
+          }, 10000);
+          
+        } else {
+          if (response.code == 200 || response.code == 201) {
+            jQuery('.response_area').text('Thank you for posting your inquiry! We will get back to you asap.');
```
Comment (ghost) [inline Bizyhood/Public/js/bizyhood-custom.js]: Let's make this text:

Thank you for your question. We will get back to you as soon as possible.

File: Bizyhood/Cta.php (hunk 1 above)
Comment (spyrosvl) [reply Bizyhood/Cta.php]: Great idea :D
Moving it!

File: Bizyhood/Cta.php (hunk 2 above)
Comment (spyrosvl) [reply Bizyhood/Cta.php]: Adding a log for it.

File: Bizyhood/Public/js/bizyhood-custom.js (hunk 3 above)
Comment (spyrosvl) [reply Bizyhood/Public/js/bizyhood-custom.js]: Changing and submitting.

**Comment by ghost** [comment]
Spyros, the Ask A Question button is not showing up on the business details page:

https://pubdemo.bizyhood.com/business-overview/abc-playschool-racine-wi-53402/3f4f3e0f-f740-4d38-b53d-17e54f2978dc/

I don't even see the markup in the page source. Please advise.

**Comment by ghost** [comment]
Looks good Spyros, thank you!

**Pull request merged**